File: qualifier-week-4/non-graded-assignemts/PPA4 - Not Graded.c
#include <stdio.h>

int main() {
    int employee_id[5];
    int i;
    for (i = 0; i < 5; i++) {
        scanf("%d", &employee_id[i]);
    }

    for (i = 0; i < 5; i++) {
        if ((employee_id[i] + employee_id[(i + 1) % 5]) % 2 != 0) {
            printf("NO");
            return 1;
        }
    }
    printf("YES");
    return 0;
}

File: qualifier-week-4/non-graded-assignemts/PPA4 - Not Graded.go
package main

import "fmt"

func meetingHappening(ids []int) bool {
	for i := range ids {
		nextIndex := (i + 1) % len(ids)
		if (ids[i]+ids[nextIndex])%2 != 0 {
			return false
		}
	}
	return true
}

func main() {
	var employeeIDs [5]int
	_, err := fmt.Scanf("%d %d %d %d %d", &employeeIDs[0], &employeeIDs[1], &employeeIDs[2], &employeeIDs[3], &employeeIDs[4])

	if err != nil {
		println("Error: Failure")
		return
	}

	if meetingHappening(employeeIDs[:]) {
		println("YES")
	} else {
		println("NO")
	}
}
